### DISCUSSION TITLE
Some Obsidian Syntax Issues

### DISCUSSION THREAD

**Discussion by AgedLace** [mkdocs-publisher/mkdocs-publisher#39] [General]
I have completed both basic, and advanced, Obsidian syntax testing based upon the Obsidian Help pages.

The first issue I ran into is that since I had already tested Obsidian callouts, I tried to embed that file into my note.  While it embedded in Obsidian, it did not do so on the locally generated web page.  

Second, All Obsidian comments are rendered on the locally generated web page.

I also ran into an issue with Obsidian's example of nesting an unordered list into an ordered list.  I'm certain this must be an Obsidian setting.  I turned off "smart indent in lists" and this did not fix my issue.  I will have to research this more on my end.

As for the Advanced Obsidian Markdown Syntax,  the MathJax examples did not render correctly on the local web page.  I don't have the time to research this more at this time.  Hopefully, later in the week.

I tried to push my latest changes to the github repo.  For some reason, VS Code is telling me I have missing or invalid credentials.  This, too, is an issue I have locally and will need more time to research this issue.  

I have the build and debug logs, but there were no errors / warnings generated, so I wont attach them here.

Once I am able to figure out why VS Code has stopped authenticating my git credentials, I will upload the repository for your review.  I am unable to do this right now.  I hope to get to it in the next day or two.

**Comment by mkusz** [comment]
@AgedLace  Can you point me to the sections of the obsidian documentation where are the elements you have mentioned described?

**Comment by AgedLace** [comment]
Here are the two Obsidian Help Pages ...

[Basic Formatting Syntax](https://help.obsidian.md/Editing+and+formatting/Basic+formatting+syntax) - This page covers the nested unordered list in an ordered list.

[Advanced Formatting Syntax ](https://help.obsidian.md/Editing+and+formatting/Advanced+formatting+syntax) - The very last section on this page deals with MathJax.

Also, my VSCode issue may be due to a recent update.  I'm also installing the latest Manjaro update that came in today.  Hopefully, VSCode will work again thereafter.

**Comment by AgedLace** [comment]
~~VSCode and Manjaro updates fixed my github missing / invalid authentication issue.~~  The Repo is [here](https://github.com/AgedLace/mkdocs-publisher).

~~CORRECTION - the initial push to the repo worked and I was able to push it to github, but when I try to push changes, I get the same error again.  Going back to researching.~~

NEW CORRECTION - Managed to fix it.  Working now.

**Comment by mkusz** [comment]
@AgedLace thank you for all the insight. I need to push v1.1.1 with current fixes for wikilinks and then I will create a proper workflow for reporting issues. Afer that we will create a separate issues per each of your finding, so it will be easier to track progess and do a faster release cycle. This will help to organize stuff and help me with some repo statistics in the backstage so the project will get some additional traction and visibility in other places like in [awesome MkDocs catalog](https://github.com/mkdocs/catalog) (project will be added just after this relese). Thanks for all of your work so far. It's very helpful.

PS. Some of your findings are easy to fix (MathJax) and some will need some hard development (like mixing unordered and ordered lists, since it's not supported by any Markdown extension).

**Comment by mkusz** [comment]
@AgedLace v1.1.1 is released. I hope you don't mind that I mentioned you in the official [blog post](https://mkusz.github.io/mkdocs-publisher/blog/v111/), since you are very active in this project as a tester and was a very crucial part of this release. In the next few days I will create issues menitoned in my previous message and close this discussion.

**Reply by AgedLace** [reply]
**@mkusz** I will check out v1.1.1 this weekend.  

No problem with respect to mentioning me in the blog post.  One thing I want to clarify on that page.  I'm not able to get mixed ordered / unordered lists to work in Obsidian yet.  So do not know that will be a problem with your project.  I hope to play around in Obsidian's settings to see if I can figure out why I'm not able to get it to work in Obsidian before I can test it with your project.

As for completing your new issue workflow - also not a problem, and there's no hurry, as I won't be able to look at this before the weekend.

I will not add any new issues until after you have created the new issue workflow.

**Reply by mkusz** [reply]
Mixed type lists are something that need to be developed (in Obsidian this works, but you have to indent one type of list vs another, like in [Obsidian basic syntax help](https://help.obsidian.md/Editing+and+formatting/Basic+formatting+syntax#Lists))

**Reply by AgedLace** [reply]
Yes, I type that example exactly into my Obsidian, and Obsidian will not allow me to change the auto-generated '3' to the '-' in order to create the mixed list.  Thus why I believe it's an Obsidian setting which is preventing me to do this.  

PS - I can cut and paste it into Obsidian note, and it works.  When I try to type it myself, it doesn't.

PPS - I found the problem.  It is the Outliner Plugin.  Even turning off the list setting didn't help.  Have disabled Outliner Plugin.

**Reply by mkusz** [reply]
The best option is:

- type 1. and start typing a text
- on enter you will jump to a new line with digit 2, etc.
- if you want to start an unordered list jump to a new line by pressing an ENTER and after a digit is displayed press TAB
- next you have to backspace and remove digit and put -(space)text
- if you want to back to ordered list, jump to a new by pressing an ENTER and then press SHIFT + TAB (indentation will be decreased)
- next you have to backspace and remove - and put digit continuation like 3.

**Reply by AgedLace** [reply]
That didn't work while the Outliner Plugin was enabled.  It's working now that I've disabled Outliner.

**Reply by mkusz** [reply]
Some time ago I also had some problems with Outliner and don't use it at all.

**Reply by mkusz** [reply]
PS. Obsidian has the same problem as MkDocs, to many plugins (it's a plus), but when you use a couple if them, they tend to have some problems. This is the reason why I decided to create this project: create a bunch of plugins, that work together as they should. It's hard and needs some effort (especially for testing, it's why I'm grateful for your support) but it's easier to maintain than reporting problems to multiple developers/plugins (sometimes it's hard to decide who should solve the problem). As for Obsidian, the best approach is to use as small number of plugins as possible to maintain the functionality we need.

**Reply by AgedLace** [reply]
Agreed.  When I create vaults for this project going forward, there will be minimal, to no, added plugins.  Will keep me from getting frustrated trying to figure out why something doesn't work.  

THANK YOU!!!

**Reply by mkusz** [reply]
There are a few very useful like:

- [Linter](https://github.com/platers/obsidian-linter) - helps a lot with yaml frontmatter
- [Templater](https://github.com/SilentVoid13/Templater) - the best plugin for templates
- [LanguageTool Integration](https://languagetool.org/insights/post/product-obsidian/) - best grammar/language correction (not only for Obsidian, very usefull for non EN people like me)

There are a few more that I want to integrate (or already integrated) like advanced tables, charts, mindmap and icons. When template repository will be created, Obsidian vault will be part of it (with those plugins).

**Reply by AgedLace** [reply]
I use both Linter and Templater,.  Will check out the Language Tool Integration!

**Reply by mkusz** [reply]
@AgedLace If you will ever use paid licence for LanguageTool Integration, don't forget to add this line `*/.obsidian/plugins/obsidian-languagetool-plugin` to yours `.gitignore`. It's need because licence is stored in one of the file from that directory and if you push it to the repo, it will be publically available and anybody can use it. If you ever do it, you need to write to LanguageTool support team to regenerate your licence. Been there, done that 😄

**Reply by AgedLace** [reply]
Thank you!!! Appreciate that very much.

**Reply by mkusz** [reply]
Hi @AgedLace 
I have added issue templates (need some polishing, but they are functional). Also, I have made a few issues from this discussion: 
- #46
- #49 (take a look at my comment inside the issue)
- #48
- #50

**Comment by mkusz** [comment]
I'm closing this discussion and let's move it to the reported issues.
